src/controllers/company.controller.js

```diff
@@ -0,0 +1,166 @@
+import crypto from "crypto";
+import bcrypt from "bcryptjs";
+import jwt from "jsonwebtoken";
+import GenerateResponse from "../utils/response.creator.js"
+import * as customerServices from "../services/customer.service.js";
+
+const RegisterCompany = async (req, res) => {
+    try {
+        const newRegistration = {
+            companyName : req.body.companyName,
+            companyId : req.body.companyId,
+            email : req.body.email,
+            password : req.body.password,
+            address : req.body.address,
+            registrationNumber : req.body.registrationNumber,
+            phoneNumber : req.body.phoneNumber,
+            ceoName : req.body.ceoName,
+            ceoContactNumber : req.body.ceoContactNumber,
+            ceoEmail : req.body.ceoEmail,
+            ceoAadhaarNumber : req.body.ceoAadhaarNumber,
+            verificationToken : crypto.randomBytes(32).toString("hex"),
+            verificationTokenExpires : Date.now() + 3600000,
+        };
+    
+        const existingCompany = await customerServices.FetchCompanyById(newRegistration.companyId);
+    
+        if(existingCompany){
+            return GenerateResponse(
+                res, 
+                400,
+                {},
+                "The Registration already Exist"
+            );
+        };
+    
+        const token = newRegistration.verificationToken;
+
+        newRegistration.password = await bcrypt.hash(newRegistration.password, parseInt(process.env.SALT));
+
+        console.log(newRegistration.password);
```

Comment:
Why are you logging this?

Reply:
There was a problem occurring forgot to remove it. Sorry!!